flink-runtime/src/test/java/org/apache/flink/runtime/io/network/api/writer/RecordWriterTest.java

```diff
@@ -196,10 +198,152 @@ public void testSerializerClearedAfterClearBuffers() throws Exception {
 	}
 
 	/**
-	 * Tests broadcasting events when no records have been emitted yet.
+	 * Tests broadcasting events when no records have been emitted yet in {@link BroadcastRecordWriter}.
 	 */
 	@Test
-	public void testBroadcastEventNoRecords() throws Exception {
+	public void testBroadcastWriterBroadcastEventNoRecord() throws Exception {
```

Comment:
I considered this way before and also preferred it actually.

But I wonder maybe not all the existing tests in `RecordWriterTest` are necessary for executing  in `BroadcastRecordWriter`, like `testClearBuffersAfterInterruptDuringBlockingBufferRequest` and `testSerializerClearedAfterClearBuffers` which I have not touched before. Then I only added the broadcast modes for the other existing tests.

I would double check and try to refactor the tests in a separate `BroadcastRecordWriterTest`.

Reply:
You can also override a base test with an empty method, marked overridden method `@Ignore` or have abstract base class, with two concrete `RecordWriterTest` and `BroadcastRecordWriterTest`. But maybe you are right, that the approach that you've taken now might be simpler/easier, so you would have to play with the code and use your own judgement here.

Reply:
Thanks for the suggestion! I would have a try and it could be better if all the existing tests are suitable for both `BroadcastRecordWriter` and `RecordWriter. :)